core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -282,21 +287,34 @@ class H2OContext private(val sparkSession: SparkSession, conf: H2OConf) extends
     LogUtil.setH2ONodeLogLevel(level)
   }
 
-  /** Stops H2O context.
-    *
-    * @param stopSparkContext stop also spark context
-    */
-  def stop(stopSparkContext: Boolean = false): Unit = synchronized {
+  private def stop(stopSparkContext: Boolean, stopJvm: Boolean): Unit = synchronized {
     if (!stopped) {
       uiUpdateThread.interrupt()
-      backend.stop(stopSparkContext)
+      if (stopSparkContext) {
+        sparkContext.stop()
+      }
+      // H2O cluster is managed by the user in case of manual start of external backend.
+      if (!(conf.isManualClusterStartUsed && conf.runsInExternalClusterMode)) {
+        H2O.orderlyShutdown()
+      }
+      if (stopJvm) {
+        H2O.exit(0)
+      }
       H2OContext.stop(this)
       stopped = true
     } else {
       logWarning("H2OContext is already stopped, this call has no effect anymore")
     }
   }
 
+  /** Stops H2O context.
+    *
+    * @param stopSparkContext stop also spark context
+    */
+  def stop(stopSparkContext: Boolean = false): Unit = {
+    stop(stopSparkContext, stopJvm = true)
```

Comment:
Shouldn't we let Spark to stop JVM?

Reply:
In ideal case yes :) However the current stop method worked in a way that it always tried to stop JVM.

Once we have the client-less approach we can get happily rid of this, however I did not want to change behaviour in this release.

Reply:
ok, np.